addons/addon-base-raas/packages/base-raas-services/lib/data-source/data-source-registration-service.js

```diff
@@ -79,6 +87,150 @@ class DataSourceRegistrationService extends Service {
     return result.studyEntity;
   }
 
+  async bulkReach(requestContext, { status }) {
```

Comment:
Can you remove all the reachability methods to a new service?  Maybe we can name the service 'StudyReachabilityService'?   This is because reachability testing is not limited to study registration.

Reply:
Will do. Thanks

Reply:
Done

Reply:
I've named it 'DataSourceReachabilityService' though, since study reachability depends on the DS Accounts being reachable first.